### PR TITLE
CodeQL vulnerability scanning

### PR DESCRIPTION
**Pull request by paolodamico**
## Changes

Runs a workflow when a PR is open to analyze for code vulnerabilities using GitHub's action CodeQL. Learn more at https://github.com/github/codeql

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] New/changed UI is decent on smartphones (viewport width around 360px)

### CODE REVIEW

File: .github/workflows/vulnerabilities.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Analyze for common vulnerabilities using GitHub's CodeQL
+#
+name: 'Vulnerability Scanning'
+
+on:
+    - pull_request
```
Comment (macobo) [inline .github/workflows/vulnerabilities.yml:6]: I think doing this at a PR level is kind of bad.

Say some new vulnerability comes out. Now suddenly all PRs start failing because of it. Once we have fixed it _in another PR_ we need to then merge in master for every single open PR to get it green.

A better model IMO would be to either run this on master periodically and e.g. create issues. WDYT?

File: requirements.txt [hunk 2]
```diff
@@ -121,7 +121,7 @@ kombu==4.6.8
     # via
     #   -r requirements.in
     #   celery
-lxml==4.6.3
+lxml==4.6.1
```
Comment (macobo) [inline requirements.txt:124]: Seems unrelated?

Comment (paolodamico) [reply requirements.txt:124]: This was a regression to test CodeQL as this version of `lxml` contains a couple of known vulnerabilities, but it didn't seem to pick them up.

File: .github/workflows/vulnerabilities.yml (hunk 1 above)
Comment (paolodamico) [reply .github/workflows/vulnerabilities.yml:6]: Yup agreed, I'm also not sure this is the best way to scan for vulnerabilities. For now, I'll enable image scanning on Docker Hub & ECR, we can keep the original issue open to explore further how best to address this.
